pysquared/hardware/burnwire/manager/burnwire.py

```diff
@@ -0,0 +1,147 @@
+import time
+
+from digitalio import DigitalInOut
+
+from ....logger import Logger
+from ....protos.burnwire import BurnwireProto
+
+"""
+Usage Example:
+
+from lib.pysquared.hardware.burnwire.manager.burnwire import BurnwireManager
+...
+
+antenna_deployment = BurnwireManger(logger, board.FIRE_DEPLOY1A, board.FIRE_DEPLOY1B, enable_logic = False)
+
+antenna_deployment.burn()
+"""
+
+
+class BurnwireManager(BurnwireProto):
+    """Class for managing burnwire ports."""
+
+    def __init__(
+        self,
+        logger: Logger,
+        enable_burn: DigitalInOut,
+        fire_burn: DigitalInOut,
+        enable_logic: bool = True,
+    ) -> None:
+        """
+        Initializes the burnwire manager class.
+
+        :param Logger logger: Logger instance for logging messages.
+        :param Digitalio enable_burn: A pin used for enabling the initial stage of a burnwire circuit.
+        :param Digitalio fire_burn: A pin used for enabling a specific burnwire port.
+        :param bool enable_logic: Boolean defining whether the burnwire load switches are enabled when True or False. Defaults to `True`.
+        """
+        self._log: Logger = logger
+        self._enable: bool = enable_logic
+
+        if enable_logic:
+            self._disable: bool = False
+        else:
+            self._disable: bool = True
+
+        self._enable_burn: DigitalInOut = enable_burn
+        self._fire_burn: DigitalInOut = fire_burn
+
+        self.number_of_attempts: int = 0
+
+    def burn(self, timeout_duration: float = 5.0, max_retries: int = 1) -> bool:
```

Comment:
```suggestion
    @with_retries(max_attempts=3, initial_delay=1)
    def burn(self, timeout_duration: float = 5.0) -> bool:
```

Could use the `with_retries` decorator for this functionality but I wonder how that would work with your safing logic. I guess it could depend on if you think this is a setting the satellite operator will want to manage or if you can take a "best guess." We've been taking a "best guess approach with other hardware initialization (always setting 3 tries so far)